cmake/sdp.cmake

```diff
@@ -49,3 +49,39 @@ function(sdp_assembly_generate hrt_srcs)
   add_dependencies(asm_gen syscall_list_h_target)
 
 endfunction()
+
+function(sdp_assembly_check hrt_srcs)
+  set(asm_check_msg "Checking if ASM files for Hard Real Time have changed.")
+
+  if(TARGET asm_check)
+    message(FATAL_ERROR "sdp_assembly_check() already called, please note that
+      sdp_assembly_check() must be called only once with a list of all source files."
+    )
+  endif()
+
+  add_custom_target(asm_check
+    COMMAND ${CMAKE_COMMAND} -D hrt_srcs=${hrt_srcs} -P ${ZEPHYR_NRF_MODULE_DIR}/cmake/sdp_asm_check.cmake
+    COMMAND_EXPAND_LISTS
+    COMMENT ${asm_check_msg}
+  )
+
+  add_dependencies(asm_check asm_gen)
+
+endfunction()
+
+function(sdp_assembly_update hrt_srcs)
+  set(asm_update_msg "Updating ASM files for Hard Real Time.")
+
+  if(TARGET asm_update)
```

Comment:
A common naming convention for such targets is `install-<something>`, not update, so perhaps: `install-sdp-asm` / `install-hrt-asm` are better names.